Library/Homebrew/os.rb

```diff
@@ -1,6 +1,6 @@
 module OS
   def self.mac?
-    /darwin/i === RUBY_PLATFORM
+    /darwin/i === RUBY_PLATFORM || ENV["HOMEBREW_EMULATE_MAC"]
```

Comment:
Try running current `master` on Linux and see how broken it is. This PR is still work-in-progress while I work out whether emulating Mac or actually using a Mac Travis worker is going to be easier.